src/node_buffer.cc

```diff
@@ -310,6 +310,70 @@ void Base64Slice(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+// bytesCopied = buffer.mask(mask, target, targetOffset, sourceStart, sourceEnd)
+void Mask(const FunctionCallbackInfo<Value> &args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  uint32_t mask32 = args[0]->Uint32Value();
+  char* mask = (char*)&mask32;
+
+  Local<Object> target = args[1]->ToObject(env->isolate());
+
+  if (!HasInstance(target))
+    return env->ThrowTypeError("second arg should be a Buffer");
+
+  ARGS_THIS(args.This())
+  size_t target_length = target->GetIndexedPropertiesExternalArrayDataLength();
+  char* target_data = static_cast<char*>(
+      target->GetIndexedPropertiesExternalArrayData());
+  size_t target_start;
+  size_t source_start;
+  size_t source_end;
+
+  CHECK_NOT_OOB(ParseArrayIndex(args[2], 0, &target_start));
+  CHECK_NOT_OOB(ParseArrayIndex(args[3], 0, &source_start));
+  CHECK_NOT_OOB(ParseArrayIndex(args[4], obj_length, &source_end));
+
+  // Copy 0 bytes; we're done
+  if (target_start >= target_length || source_start >= source_end)
+    return args.GetReturnValue().Set(0);
+
+  if (source_start > obj_length)
+    return env->ThrowRangeError("out of range index");
+
+  if (source_end - source_start > target_length - target_start)
+    source_end = source_start + target_length - target_start;
```

Comment:
Ignore my previous comment.  This calculation can in theory overflow (although not yet in practice due to size restrictions of buffers) assuming 32 bits size_t: `source_end = (2**31+1) + 2**31 - 0;  // 1`

I'd probably add a `CHECK_LE(source_start, source_end)`, just in case.